reVX/least_cost_xmission/trans_cap_costs.py

```diff
@@ -1087,13 +1111,158 @@ def run(cls, cost_fpath, sc_point, features, capacity_class, radius=None,
                          .format(tcc.sc_point_gid, time.time() - ts))
         except InvalidMCPStartValueError as ex:
             features = None
-            msg = ('Could not connect SC point {} to tranmission features: {}'
+            msg = ('Could not connect SC point {} to transmission features: {}'
                    .format(sc_point['sc_point_gid'], ex))
             logger.debug(msg)
         except Exception as ex:
-            msg = ('Failed to connect SC point {} to tranmission features: {}'
+            msg = ('Failed to connect SC point {} to transmission features: {}'
                    .format(sc_point['sc_point_gid'], ex))
             logger.exception(msg)
             raise
 
         return features
+
+
+class ReinforcementLineCosts(TieLineCosts):
+    """
+    Compute Least Cost Reinforcement Line cost from substations to
+    network nodes.
+
+    The reinforcement line path will attempt to follow existing
+    transmission lines for as long as possible. Costs are calculated
+    using half of the greenfield cost of the transmission line that is
+    being traced. If the reinforcement path travels along two different
+    line voltages, corresponding costs are used for each portion of the
+    path. In the case that the path must cross a region with no existing
+    transmission lines to reach the destination, half (50%) of the
+    greenfield cost of the input ``capacity_class`` is used.
+    """
+    def __init__(self, transmission_lines, cost_fpath, start_idx,
+                 capacity_class, row_slice, col_slice, xmission_config=None,
+                 barrier_mult=100):
+        """
+
+        Parameters
+        ----------
+        transmission_lines : dict
+            Dictionary where the keys are the names of cost layers in
+            the cost HDF5 file and values are arrays with the
+            corresponding existing transmission lines rastered into
+            them (i.e. array value is 1 at a pixel if there is a
+            transmission line, otherwise 0). These arrays will be used
+            to compute the reinforcement costs along existing
+            transmission lines of differing voltages.
+        cost_fpath : str
+            Full path of .h5 file with cost arrays.
+        start_idx : tuple
+            Tuple of (row_idx, col_idx) to compute reinforcement line
+            path to.
```

Comment:
This is just a general point of confusion on `start_idx` and `end_indices`. So these are the endpoint indexes in the `transmission_lines` rasters to compute cost paths between? Can we say "`(row_idx, col_idx) in the transmission_lines arrays`" (e.g. explicitly point back to the transmission lines input)? Both docstrings say "path to"... could we say "from start to end" or something similar so their relationship is explicit? How do we feel about naming them consistently, e.g. `start_indices` and `end_indices`?

Reply:
I think renaming to `start_indices` and `end_indices` is a good idea. Implemented that below. Also added quite a bit of elaboration in the docstrings as well. Let me know if there is still any confusion!